sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CheckpointFileManager.scala

```diff
@@ -160,11 +160,16 @@ object CheckpointFileManager extends Logging {
     override def cancel(): Unit = synchronized {
       try {
         if (terminated) return
-        underlyingStream.close()
+        try {
```

Comment:
do we need to fix `close()` method as well since seems it also doesn't handle IO exception from `underlyingStream.close()`?

Reply:
`cancel` is called to abandon the temporary output, so if it is just non fatal exception, we don't handle it.

Reply:
Oh misreading your question. `close` cannot avoid the IO exception from `underlyingStream.close`. So it doesn't catch it. Such exception will fail the streaming query.

Reply:
Ah I see. Thanks.